# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: third_party/rust/wgpu-core/src/command/pass.rs
# Commit: 3f167f8b3137
# Full Hash: 3f167f8b313749f50f135e189e5854bef7ac0c73
# Author: Erich Gubler <erichdongubler@gmail.com>
# Date: 2025-10-03 18:08 +0000
# Regressor Bug: 1991551
# File Overlap Count: 23
# Overlapping Files: third_party/rust/wgpu-hal/Cargo.toml, supply-chain/audits.toml, third_party/rust/naga/src/back/spv/writer.rs, third_party/rust/naga/src/back/msl/writer.rs, third_party/rust/wgpu-types/src/lib.rs
# Description:
#   Bug 1991551 - build(webgpu): update WGPU to 8c4aebc0c1b75e3f7412dfc180745a0717fc30ec r=webgpu-reviewers,supply-chain-reviewers,nical
# ==============================================================================

diff -r 823439f307a0 -r 3f167f8b3137 third_party/rust/wgpu-core/src/command/pass.rs
--- a/third_party/rust/wgpu-core/src/command/pass.rs	Fri Oct 03 18:01:30 2025 +0000
+++ b/third_party/rust/wgpu-core/src/command/pass.rs	Fri Oct 03 18:08:29 2025 +0000
@@ -4,8 +4,8 @@
 use crate::command::bind::Binder;
 use crate::command::encoder::EncodingState;
 use crate::command::memory_init::SurfacesInDiscardState;
-use crate::command::{CommandEncoder, DebugGroupError, QueryResetMap, QueryUseError};
-use crate::device::{DeviceError, MissingFeatures};
+use crate::command::{DebugGroupError, QueryResetMap, QueryUseError};
+use crate::device::{Device, DeviceError, MissingFeatures};
 use crate::pipeline::LateSizedBufferGroup;
 use crate::ray_tracing::AsAction;
 use crate::resource::{DestroyedResourceError, Labeled, ParentDevice, QuerySet};
@@ -41,8 +41,8 @@
     }
 }
 
-pub(crate) struct PassState<'scope, 'snatch_guard, 'cmd_enc, 'raw_encoder> {
-    pub(crate) base: EncodingState<'snatch_guard, 'cmd_enc, 'raw_encoder>,
+pub(crate) struct PassState<'scope, 'snatch_guard, 'cmd_enc> {
+    pub(crate) base: EncodingState<'snatch_guard, 'cmd_enc>,
 
     /// Immediate texture inits required because of prior discards. Need to
     /// be inserted before texture reads.
@@ -61,7 +61,7 @@
 
 pub(crate) fn set_bind_group<E>(
     state: &mut PassState,
-    cmd_enc: &CommandEncoder,
+    device: &Arc<Device>,
     dynamic_offsets: &[DynamicOffset],
     index: u32,
     num_dynamic_offsets: usize,
@@ -108,7 +108,7 @@
     let bind_group = bind_group.unwrap();
     let bind_group = state.base.tracker.bind_groups.insert_single(bind_group);
 
-    bind_group.same_device_as(cmd_enc)?;
+    bind_group.same_device(device)?;
 
     bind_group.validate_dynamic_bindings(index, &state.temp_offsets)?;
 
@@ -271,7 +271,7 @@
 
 pub(crate) fn write_timestamp<E>(
     state: &mut PassState,
-    cmd_enc: &CommandEncoder,
+    device: &Arc<Device>,
     pending_query_resets: Option<&mut QueryResetMap>,
     query_set: Arc<QuerySet>,
     query_index: u32,
@@ -284,7 +284,7 @@
         query_set.error_ident()
     );
 
-    query_set.same_device_as(cmd_enc)?;
+    query_set.same_device(device)?;
 
     state
         .base